UIDS-GIDS.md

```diff
@@ -0,0 +1,246 @@
+# Users, Groups, UIDs and GIDs on `systemd` systems
+
+Here's a summary of the requirements `systemd` (and Linux) make on UID/GID
+assignments and their ranges.
+
+Note that while in theory UIDs and GIDs are orthogonal concepts they really
+aren't IRL. With that in mind, when we discuss UIDs below it should be assumed
+that whatever we say about UIDs applies to GIDs in mostly the same way, and all
+the special assignments and ranges for UIDs always have mostly the same
+validity for GIDs too. To say this clearly: every allocated system user should
+have a matching group, and they should share the same numeric value for their
+UIDs and GIDs. That said, while users/UIDs generally should have matching
```

Comment:
UID!=GID is true for all debian-based systems. The `base-files` database comes with a `user` group (GID=100), which comes of course with no associated user. Since UIDs and GIDs are then picked in order, all system users created dynamically will have UID!=GID.

```
>>> import pwd
>>> len(pwd.getpwall())
36
>>> len([p for p in pwd.getpwall() if p.pw_uid != p.pw_gid])
19
```

Reply:
well, the text says "should"... it doesn't say "it's broken if not that way"...

I mean, I think its a very poor choice if the UIDs and GIDs don't match there... I mean, we can deal with it, but I think its not a good idea, as it suggests allocation of system uids/gids was really orthogonal, but it really is not...